spinn_front_end_common/utility_models/data_speed_up_packet_gatherer_machine_vertex.py

```diff
@@ -0,0 +1,754 @@
+                "seq num {} length of packet {} and final {}".format(
+                    len(self._output), view_end_position, seq_num,
+                    self._max_seq_num, packet_length, is_final))
+        # print "view_start={} view_end={} data_start={} data_end={}".format(
+        # view_start_position, view_end_position, data_start_position,
+        # data_end_position)
+        self._view[view_start_position: view_end_position] = \
+            data[data_start_position:data_end_position]
+
+    def _check(self, seq_nums):
+        """ verifying if the seq nums are correct.
+
+        :param seq_nums: the received seq nums
+        :return: bool of true or false given if all the seq nums been received
+        """
+        # hand back
+        seq_nums = sorted(seq_nums)
+        max_needed = self.calculate_max_seq_num()
+        if len(seq_nums) > max_needed:
+            raise Exception(
+                "I've received more data than i was expecting!!")
+        if len(seq_nums) != max_needed:
+            # self._print_length_of_received_seq_nums(seq_nums, max_needed)
+            return False
+        return True
+
+    def calculate_max_seq_num(self):
+        """ deduces the max seq num expected to be received
+
+        :return: int of the biggest seq num expected
+        """
+        n_sequence_numbers = 0
+        data_left = len(self._output) - (
+            (self.DATA_PER_FULL_PACKET -
+             self.SDP_RETRANSMISSION_HEADER_SIZE) *
+            self.WORD_TO_BYTE_CONVERTER)
+
+        extra_n_sequences = float(data_left) / float(
+            self.DATA_PER_FULL_PACKET_WITH_SEQUENCE_NUM *
+            self.WORD_TO_BYTE_CONVERTER)
+        n_sequence_numbers += math.ceil(extra_n_sequences)
+        return int(n_sequence_numbers)
+
+    @staticmethod
+    def _print_missing(seq_nums):
+        """ debug printer for the missing seq nums from the pile
+
+        :param seq_nums: the seq nums received so far
+        :rtype: None
+        """
+        last_seq_num = 0
+        seq_nums = sorted(seq_nums)
+        for seq_num in seq_nums:
+            if seq_num != last_seq_num + 1:
+                print "from list im missing seq num {}".format(seq_num)
+            last_seq_num = seq_num
+
+    def _print_out_packet_data(self, data):
+        """ debug prints out the data from the packet
+
+        :param data: the packet data
+        :rtype: None
+        """
+        reread_data = struct.unpack("<{}I".format(
+            int(math.ceil(len(data) / self.WORD_TO_BYTE_CONVERTER))),
+            str(data))
+        print "converted data back into readable form is {}" \
+            .format(reread_data)
+
+    @staticmethod
+    def _print_length_of_received_seq_nums(seq_nums, max_needed):
+        """ debug helper method for figuring out if everything been received
+
+        :param seq_nums: seq nums received
+        :param max_needed: biggest expected to have
+        :rtype: None
```

Comment:
You return `False` just below!

Reply:
copy and paste error when was moving prints outside the flow into their own methods. remvoed

Reply:
It did indeed look like a copy-and-paste error, but thanks for confirming!